proxy/src/main/java/com/velocitypowered/proxy/connection/client/StatusSessionHandler.java

```diff
@@ -41,10 +45,32 @@ private ServerPing createInitialPing() {
             ImmutableList.of()),
         configuration.getMotdComponent(),
         configuration.getFavicon().orElse(null),
-        configuration.isAnnounceForge() ? ModInfo.DEFAULT : null
+        getModInfo(configuration.isAnnounceForge())
     );
   }
 
+  private ModInfo getModInfo(boolean isAnnounceForge) {
+    for (String serverName : server.getConfiguration().getAttemptModInfoOrder()) {
+      Optional<RegisteredServer> registeredServer = server.getServer(serverName);
+      if (!registeredServer.isPresent()) {
+        continue;
+      }
+      try {
+        ServerPing serverPing = registeredServer.get().ping().get();
```

Comment:
Big red flag. This is a blocking operation that takes place on a Netty I/O thread. That's a no-no.